recipes/dcv_install.rb

```diff
@@ -66,21 +73,41 @@ def disable_lock_screen
   not_if { ::File.exist?("#{node['cfncluster']['scripts_dir']}/pcluster_dcv_connect.sh") }
 end
 
-
-if node['platform'] == 'centos' && node['platform_version'].to_i == 7 && !File.exist?("/etc/dcv/dcv.conf")
+if node['cfncluster']['dcv']['supported_os'].include?("#{node['platform']}#{node['platform_version'].to_i}") && !File.exist?("/etc/dcv/dcv.conf")
   case node['cfncluster']['cfn_node_type']
   when 'MasterServer', nil
     dcv_tarball = "#{node['cfncluster']['sources_dir']}/dcv-#{node['cfncluster']['dcv']['version']}.tgz"
 
-    # Install the desktop environment and the desktop manager packages
-    execute 'Install gnome desktop' do
-      command 'yum -y install @gnome'
+    # Install DCV pre-requisite packages
+    case node['platform']
+    when 'centos'
+      # Install the desktop environment and the desktop manager packages
+      execute 'Install gnome desktop' do
+        command 'yum -y install @gnome'
+      end
+      # Install X Window System (required when using GPU acceleration)
+      package "xorg-x11-server-Xorg"
+
+    when 'ubuntu'
+      # Install the desktop environment and the desktop manager packages
+      # Must purge ifupdown before creating the AMI or the instance will have an ssh failure
+      bash 'install pre-req' do
+        user 'root'
+        group 'root'
+        cwd Chef::Config[:file_cache_path]
+        code <<-PREREQ
+          set -e
+          apt -y install ubuntu-desktop
+          apt -y purge ifupdown
+          DEBIAN_FRONTEND=noninteractive apt -y install lightdm
+          apt -y install mesa-utils
+          wget https://d1uj6qtbmh3dt5.cloudfront.net/NICE-GPG-KEY
+          gpg --import NICE-GPG-KEY
+        PREREQ
+      end
     end
     disable_lock_screen
```

Comment:
please double check if the lock screen disable works for ubuntu.

Reply:
The same procedure works for Centos and Ubuntu: https://askubuntu.com/a/1184810
Also confirming that lock screen is disabled with manual testing